### PR TITLE
Added `nonlocal`. `global` takes multiple args.

### PR DESCRIPTION
**Pull request by unmerged**
Related to https://github.com/hylang/hy/issues/246

### CODE REVIEW

File: tests/compilers/test_ast.py [hunk 1]
```diff
@@ -211,12 +211,25 @@ def test_ast_bad_assert():
...
 
 
 def test_ast_bad_global():
     "Make sure AST can't compile invalid global"
     cant_compile("(global)")
-    cant_compile("(global foo bar)")
+    cant_compile("(global (foo))")
+
+
+if PY3:
```
Comment (paultag) [inline tests/compilers/test_ast.py]: so skipping won't work great if we raise a SyntaxError -- which is why we have the if-python2 macro

**Comment by farhaven** [comment]
The tests seem to run fine here with both Python 2.7 and 3.4. This is based on a git HEAD from today with cfa805c and 41e5175 cherry-picked in.

**Comment by Foxboron** [comment]
All tests are passing with this patch. LGTM.
Then we could also get the #803 merged
:shipit:

**Comment by paultag** [comment]
LGTM; this is awesome! Nice work, @unmerged !

**Pull request merged**